src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -867,23 +867,36 @@ public virtual int GetByteCount(char[] chars)
         }
 
         [Pure]
-        public virtual int GetByteCount(String s)
-        {
-            if (s==null)
-                throw new ArgumentNullException(nameof(s));
-            Contract.EndContractBlock();
-
-            char[] chars = s.ToCharArray();
-            return GetByteCount(chars, 0, chars.Length);
-
-        }
+        public virtual int GetByteCount(String s) => GetByteCount(s, 0, s.Length);
```

Comment:
The existing `GetByteCount(string)` has to stay as as. It is a breaking change to change it to call the new GetByteCount.

Evolving classes with overriable virtual methods in compatible way is complex...